### PR TITLE
[Hotfix]Fix pf flow build executable

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py [hunk 1]
```diff
@@ -647,23 +646,51 @@ def get_git_base_dir():
...
+        required_packages = get_package_dependencies(dependencies)
+        # hard-code promptflow-** required extra packages for now
+        extra_packages = [
+            "pyarrow",
+            "pyinstaller",
+            "streamlit",
+            "streamlit-quill",
+            "bs4",
+            "fastapi",
+            "azure-identity",
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py:688]: why do we need the azure ones?

Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py:688]: it's promptflow core extra dependency。 Not sure whether we need this in executable
image

File: src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py [hunk 2]
```diff
@@ -182,16 +182,13 @@ def is_pfs_service_healthy(pfs_port) -> bool:
...
     time.sleep(time_delay)
     is_healthy = is_pfs_service_healthy(pfs_port)
-    while is_healthy is False and time_threshold > wait_time:
-        logger.info(
-            f"Promptflow service is not ready. It has been waited for {wait_time}s, will wait for at most "
-            f"{time_threshold}s."
-        )
-        wait_time += time_delay
+    while is_healthy is False and count_threshold > cnt:
+        logger.info(f"Promptflow service is not ready. will try at most {count_threshold} times.")
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:190]: print the cnt here?

File: src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py [hunk 3]
```diff
@@ -17,7 +17,6 @@
 from typing import Dict, Iterable, List, NoReturn, Tuple, Union
 
 import pydash
-from pip._vendor import tomli as toml
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py:20]: + @crazygao @brynn-code for review

File: src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py [hunk 4]
```diff
@@ -647,23 +646,51 @@ def get_git_base_dir():
...
 
+        def get_package_dependencies(package_name_list):
+            dependencies = []
+            for package_name in package_name_list:
+                if is_tool("conda"):
+                    result = subprocess.run(
+                        "conda activate root | pip show {}".format(package_name), shell=True, stdout=subprocess.PIPE
+                    )
+                else:
+                    result = subprocess.run(["pip", "show", package_name], stdout=subprocess.PIPE)
```
Comment (crazygao) [inline src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py:665]: pip/conda is not guaranteed.

File: src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py (hunk 2 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_service/utils/utils.py:190]: done

File: src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py (hunk 4 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py]: offline synced, will use a separate txt file to hold packages for executable. And add a test to keep the file consistent with toml file of devkit/core/tracing

File: src/promptflow-devkit/promptflow/_sdk/data/executable/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,39 @@
+openai
```
Comment (crazygao) [inline src/promptflow-devkit/promptflow/_sdk/data/executable/requirements.txt:1]: This file is not compiled.

File: src/promptflow/tests/sdk_cli_test/unittests/test_utils.py [hunk 6]
```diff
@@ -488,3 +490,43 @@ def test_gen_uuid_by_compute_info(self):
...
+            dependencies = [
+                dependency
+                for dependency in dependencies
+                if not dependency.startswith("promptflow") and not dependency == "python"
+            ]
+            return dependencies
+
+        all_packages = get_toml_dependencies()
+
+        with open(
```
Comment (wangchao1230) [inline src/promptflow/tests/sdk_cli_test/unittests/test_utils.py:520]: add more comments here, so devs know what to do when they see fails

File: src/promptflow-devkit/promptflow/_sdk/data/executable/requirements.txt (hunk 5 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/data/executable/requirements.txt:1]: offline synced and verified

File: src/promptflow/tests/sdk_cli_test/unittests/test_utils.py (hunk 6 above)
Comment (YingChen1996) [reply src/promptflow/tests/sdk_cli_test/unittests/test_utils.py:520]: done

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**